### PR TITLE
[Core] Remove some deprecated compilation warnings

### PR DESCRIPTION
**Pull request by loumalouomega**
**Description**
Remove some deprecated compilation warnings

**Changelog**
- Replace NumThreads 
- Replace timer 
- Replace macro _OMP

### CODE REVIEW

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h [hunk 1]
```diff
@@ -1273,7 +1277,7 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
         // Final multiplication
         TSparseSpace::Mult(T_transpose_matrix, rb_copy, rb);
 
-        const double stop_build = OpenMPUtils::GetCurrentTime();
+        const double stop_build = timer.ElapsedSeconds();
         KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Constraint relation build time and multiplication: " << stop_build - start_build << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h]:
```suggestion
        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Constraint relation build time and multiplication: " << timer.ElapsedSeconds()<< std::endl;
```

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h [hunk 2]
```diff
@@ -1230,7 +1233,8 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
         KRATOS_TRY
 
         // Assemble the constraints
-        const double start_build = OpenMPUtils::GetCurrentTime();
+        const auto timer = BuiltinTimer();
+        const double start_build = timer.ElapsedSeconds();
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h]:
```suggestion
```

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h [hunk 3]
```diff
@@ -1273,7 +1277,7 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
         // Final multiplication
         TSparseSpace::Mult(T_transpose_matrix, rb_copy, rb);
 
-        const double stop_build = OpenMPUtils::GetCurrentTime();
+        const double stop_build = timer.ElapsedSeconds();
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h]:
```suggestion
```

**Review by philbucher — CHANGES_REQUESTED** [review]
you are using the timer wrong, it is easier than before

Otherwise ok thx

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h (hunk 3 above)
Comment (loumalouomega) [reply kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h]: OK, understood

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 4]
```diff
@@ -272,8 +271,7 @@ class ResidualBasedBlockBuilderAndSolver
             }
         }
 
-        const double stop_build = timer.ElapsedSeconds();
-        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Build time: " << stop_build - start_build << std::endl;
+        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Build time: " << timer.ElapsedSeconds() << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]:
```suggestion
        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", this->GetEchoLevel() >= 1) << "Build time: " << timer.ElapsedSeconds() << std::endl;

```

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 5]
```diff
@@ -534,14 +530,12 @@ class ResidualBasedBlockBuilderAndSolver
...
         const auto timer = BuiltinTimer();
-        const double start_solve = timer.ElapsedSeconds();
         Timer::Start("Solve");
 
         SystemSolveWithPhysics(A, Dx, b, rModelPart);
 
         Timer::Stop("Solve");
-        const double stop_solve = timer.ElapsedSeconds();
-        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", (this->GetEchoLevel() >=1 && rModelPart.GetCommunicator().MyPID() == 0)) << "System solve time: " << stop_solve - start_solve << std::endl;
+        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", (this->GetEchoLevel() >=1 && rModelPart.GetCommunicator().MyPID() == 0)) << "System solve time: " << timer.ElapsedSeconds() << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]:
```suggestion
        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", this->GetEchoLevel() >=1) << "System solve time: " << timer.ElapsedSeconds() << std::endl;

```

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 6]
```diff
@@ -669,14 +663,12 @@ class ResidualBasedBlockBuilderAndSolver
...
         const auto timer = BuiltinTimer();
-        const double start_solve = timer.ElapsedSeconds();
         Timer::Start("Solve");
 
         SystemSolveWithPhysics(rA, rDx, rb, rModelPart);
 
         Timer::Stop("Solve");
-        const double stop_solve = timer.ElapsedSeconds();
-        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", (this->GetEchoLevel() >=1 && rModelPart.GetCommunicator().MyPID() == 0)) << "System solve time: " << stop_solve - start_solve << std::endl;
+        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", (this->GetEchoLevel() >=1 && rModelPart.GetCommunicator().MyPID() == 0)) << "System solve time: " << timer.ElapsedSeconds() << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]:
```suggestion
        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolver", this->GetEchoLevel() >=1) << "System solve time: " << timer.ElapsedSeconds() << std::endl;

```

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h [hunk 7]
```diff
@@ -261,8 +260,7 @@ class ResidualBasedEliminationBuilderAndSolver
                 }
             }
         }
-        const double stop_build = timer.ElapsedSeconds();
-        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolver", (this->GetEchoLevel() >=1 && rModelPart.GetCommunicator().MyPID() == 0)) << "System build time: " << stop_build - start_build << std::endl;
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolver", (this->GetEchoLevel() >=1 && rModelPart.GetCommunicator().MyPID() == 0)) << "System build time: " << timer.ElapsedSeconds() << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h]:
```suggestion
        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolver", this->GetEchoLevel() >=1) << "System build time: " << timer.ElapsedSeconds() << std::endl;

```

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h [hunk 8]
```diff
@@ -261,8 +260,7 @@ class ResidualBasedEliminationBuilderAndSolver
                 }
             }
         }
-        const double stop_build = timer.ElapsedSeconds();
-        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolver", (this->GetEchoLevel() >=1 && rModelPart.GetCommunicator().MyPID() == 0)) << "System build time: " << stop_build - start_build << std::endl;
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolver", (this->GetEchoLevel() >=1 && rModelPart.GetCommunicator().MyPID() == 0)) << "System build time: " << timer.ElapsedSeconds() << std::endl;
 
         KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolver", this->GetEchoLevel() > 2 && rModelPart.GetCommunicator().MyPID() == 0) << "Finished building" << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h]:
```suggestion
        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolver", this->GetEchoLevel() > 2) << "Finished building" << std::endl;

```

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h [hunk 9]
```diff
@@ -547,14 +545,12 @@ class ResidualBasedEliminationBuilderAndSolver
...
         const auto timer = BuiltinTimer();
-        const double start_solve = timer.ElapsedSeconds();
         Timer::Start("Solve");
 
         SystemSolveWithPhysics(rA, rDx, rb, rModelPart);
 
         Timer::Stop("Solve");
-        const double stop_solve = timer.ElapsedSeconds();
-        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolver", (this->GetEchoLevel() >=1 && rModelPart.GetCommunicator().MyPID() == 0)) << "System solve time: " << stop_solve - start_solve << std::endl;
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolver", (this->GetEchoLevel() >=1 && rModelPart.GetCommunicator().MyPID() == 0)) << "System solve time: " << timer.ElapsedSeconds() << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h]:
```suggestion
        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolver", this->GetEchoLevel() >=1) << "System solve time: " << timer.ElapsedSeconds() << std::endl;

```

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h [hunk 10]
```diff
@@ -1277,8 +1274,7 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
         // Final multiplication
         TSparseSpace::Mult(T_transpose_matrix, rb_copy, rb);
 
-        const double stop_build = timer.ElapsedSeconds();
-        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Constraint relation build time and multiplication: " << stop_build - start_build << std::endl;
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Constraint relation build time and multiplication: " << timer.ElapsedSeconds() << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h]:
```suggestion
        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", this->GetEchoLevel() >= 1) << "Constraint relation build time and multiplication: " << timer.ElapsedSeconds() << std::endl;
```

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h [hunk 11]
```diff
@@ -1209,8 +1208,7 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
         auxiliar_A_matrix.resize(0, 0, false);
         T_transpose_matrix.resize(0, 0, false);
 
-        const double stop_build = timer.ElapsedSeconds();
-        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Constraint relation build time and multiplication: " << stop_build - start_build << std::endl;
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Constraint relation build time and multiplication: " << timer.ElapsedSeconds() << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h]:
```suggestion
        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", this->GetEchoLevel() >= 1) << "Constraint relation build time and multiplication: " << timer.ElapsedSeconds() << std::endl;
```

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h [hunk 12]
```diff
@@ -1209,8 +1208,7 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
...
 
         KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", (this->GetEchoLevel() > 2 && rModelPart.GetCommunicator().MyPID() == 0)) << "Finished parallel building with constraints" << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h]:
```suggestion
        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", this->GetEchoLevel() > 2) << "Finished parallel building with constraints" << std::endl;
```

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h [hunk 13]
```diff
@@ -1277,8 +1274,7 @@ class ResidualBasedEliminationBuilderAndSolverWithConstraints
         // Final multiplication
         TSparseSpace::Mult(T_transpose_matrix, rb_copy, rb);
 
-        const double stop_build = timer.ElapsedSeconds();
-        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Constraint relation build time and multiplication: " << stop_build - start_build << std::endl;
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Constraint relation build time and multiplication: " << timer.ElapsedSeconds() << std::endl;
 
         KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", (this->GetEchoLevel() > 2 && rModelPart.GetCommunicator().MyPID() == 0)) << "Finished parallel building with constraints" << std::endl;
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h]:
```suggestion
        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", this->GetEchoLevel() > 2) << "Finished parallel building with constraints" << std::endl;
```

**Review by philbucher — COMMENTED** [review]
minor

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h (hunk 4 above)
Comment (RiccardoRossi) [reply kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: guys, why are u taking out the mpi related stuff??

**Review by RiccardoRossi — DISMISSED** [review]
guys, the mpi-related changes should not be removed...

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h (hunk 4 above)
Comment (philbucher) [reply kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]: This is a leftover, the logger handles this

**Comment by loumalouomega** [comment]
> guys, the mpi-related changes should not be removed...

I have not touched them, what do I do then? @RiccardoRossi @philbucher

**Comment by philbucher** [comment]
> > guys, the mpi-related changes should not be removed...
> 
> I have not touched them, what do I do then? @RiccardoRossi @philbucher

I explained it to @RiccardoRossi but now he is not answering xD
(I am right, I know it :D)

**Comment by RiccardoRossi** [comment]
if you are 200% sure i will remove my blocker

**Comment by philbucher** [comment]
For the record, brief summary of the discussion I had with @RiccardoRossi :
- `KRATOS_INFO...` prints only on rank zero => additional checks for rank zero are superfluous (see my suggestions above)
- I.e. the following will never print anything (only relevant for MPI): `KRATOS_INFO_IF(my_pid_==1)`
- for printing on all ranks (also only relevant for MPI) we have the `_ALL_RANKS` macros:
https://github.com/KratosMultiphysics/Kratos/blob/74ca63383d1341c849e89554202582b4326d02fb/kratos/input_output/logger.h#L258-L261
see e.g. https://github.com/KratosMultiphysics/Kratos/blob/74ca63383d1341c849e89554202582b4326d02fb/applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h#L602-L605

related: #5863, #6615

**Review by philbucher — APPROVED** [review]
thx

**Pull request merged**